src/components/ListLayout.jsx

```diff
@@ -61,6 +61,94 @@ const ListLayout = ({ items, localToken }) => {
     return timeCheck;
   }
 
+  // updates isActive property of item to true if item has 2+ purchases and has been purchased within calculated estimate
+  // isActive is defaulted to false when item is added
+  items.forEach((item) => {
+    const dateOfLastTransaction =
+      item.totalPurchases > 0 ? item.purchasedDate : item.createdAt;
+    const daysSinceLastTransaction =
+      (currentTime - dateOfLastTransaction) / oneDay;
+    if (
+      item.totalPurchases > 1 &&
+      daysSinceLastTransaction < 2 * item.previousEstimate
+    ) {
+      let itemToUpdate = {
+        ...item,
+        isActive: true,
+      };
+      setUpdateToDb(localToken, item.id, itemToUpdate);
+    }
+  });
+
+  //this sorts by previousEstimate (calculation of when user will buy the item again) Note: items is already sorted alphabetically by item Id which is the normalized item name
+  items
+    .sort((itemA, itemB) => itemA.previousEstimate - itemB.previousEstimate)
+    .sort((itemA, itemB) => Number(itemB.isActive) - Number(itemA.isActive));
+  let frequencyGroup;
+  function colorClass(item) {
+    //inactive = gray
+    if (!item.isActive) {
+      frequencyGroup = 'inactive';
+      return 'bg-gray-300';
+    }
+
+    //buy soon
+    else if (item.previousEstimate < 7) {
```

Comment:
For the active items, this might be a good place to use a [switch statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch). 

Another thing you might want to consider is updating the `frequencyGroup` within the context of an item as a property as opposed to updating the global variable (in terms of the component).
Ex:
```
if (!item.isActive) {
      item.frequencyGroup = 'inactive';
      return 'bg-gray-300';
    }
```

Outside of the context of the item, `frequencyGroup` is almost meaningless, so it would be nice to represent that more clearly in the code. The property value may also be more useful in the future (one thing that comes to mind is debugging).

Reply:
I also recognize that updating the item like this `item.frequencyGroup` is kind of against what I said in my other comment about since this is directly updating the `items`. Just food for thought. @EdwardHinkle & @rudidev08 - do you have any other insight about this?